### DISCUSSION TITLE
Will ABP have to do something to get performance benefit?

### DISCUSSION THREAD

**Discussion by sturlath** [Megabit/Blazorise#4291] [General]
I just saw this [discussion ](https://github.com/Megabit/Blazorise/pull/4103#issuecomment-1250074828) and noticed some huge performance gains there but then saw the following comment

> Because

> **ABP** has it's IComponentActivator
> They're using Blazorise for Blazor implementation
> And with IComponentDisposer we've only focused on Microsoft DI and ServiceProviderEngineScope implementation. In the ABP > framework, you as a user can choose to form multiple different DI libraries. Meaning our IComponentDisposer will not work in all > cases.

So, **my question** is whether ABP will get them out of the box with [their update to 1.1.3](https://github.com/abpframework/abp/issues/14589) or if something needs to be done?

I´m trying to find anything that can [make ABP.io WASM faster](https://github.com/abpframework/abp/issues/13114) and stumbled on this issue when trying to figure out some `Document is not focused.` `at ee line null, column null (null:null)` (I´m trying to log more information to know where this comes from so I´m not 100% if my issue is related to this PR)

**Comment by David-Moreira** [comment]
It should be on by default if the default DI is used. 
Unless apb has specifically overriden our implementation.

@stsrki do you have more concrete info about this?

**Comment by stsrki** [comment]
Yeah, in ABP, they have their own [IComponentActivator](https://github.com/abpframework/abp/blob/e3e1779de6df5d26f01cdc8e99ac9cbcb3d24d3c/framework/src/Volo.Abp.AspNetCore.Components/Volo/Abp/AspNetCore/Components/DependencyInjection/ServiceProviderComponentActivator.cs). It is basically a default implementation that on it's doesn't do much. My advice would be to override it with Blazorise for the best performance.

**Reply by sturlath** [reply]
Is it straight forward for me to override it? How would you do it, you know?

**Reply by David-Moreira** [reply]
I think @sturlath is talking about the new IComponentDisposer, but it won't do miracles performance wise.
It's mainly for memory clean up / mem leaks.

**Reply by David-Moreira** [reply]
Basically you should see how fast Blazorise is without apb for your use case and compare... 
It might be something on apb...

**Reply by sturlath** [reply]
I have been experiencing come mem leaks so I would love to have it. It would surprise me if abp.io wouldn´t be interested getting that. And yes, Blazorise without abp is much more performant at the moment (even a bare bone apb version)... and WASM doesn´t have much priority it seems.. so I´m trying to search for everything to make it perform little better..

**Reply by stsrki** [reply]
We register services as scoped, while ABP registers them as transient. See for reference https://github.com/Megabit/Blazorise/blob/master/Source/Blazorise/Config.cs#L30

It should be enough if you just re-register IComponentActivator as scoped, BUT sometime after the ABP register it with their modules import.

Eg.

`services.Replace(ServiceDescriptor.Scoped<IFoo, FooB>());`

**Reply by sturlath** [reply]
OK I´ll try this after we have updated the abp next month..

**Reply by sturlath** [reply]
Just wanted to add [a link to the performance issue](https://github.com/abpframework/abp/issues/13114#issuecomment-1312775297) at abp for those that are interested.
